testing/adios2/python/TestADIOS.py

```diff
@@ -0,0 +1,56 @@
+from adios2.adios import ADIOS
+import unittest
+
+
+class TestADIOS(unittest.TestCase):
```

Comment:
I noticed in the `pyproject.toml` that you mention `pytest` as an optional dependency, but here are using the built-in `unittest` module instead.  In my experience `pytest` is preferred these days, and the [documentation](https://learn.scientific-python.org/development/guides/pytest/) I've been following for the pip packaging PR recommends using it.

Reply:
hmm actually we do not need pytest we run the tests with ctest in the light of it it might be better to use the builtin module to avoid using external pkgs. What do you think?

Reply:
Here's a [comparison](https://builtin.com/data-science/pytest-vs-unittest) of `unittest` vs `pytest` I found.  It basically says `unittest` is fine if you can't install dependencies for some reason, but `pytest` is better in every other way.  And in our case, I'm not sure I see the value of avoiding external packages, especially since we already depend on `numpy`, and the pip packaging PR introduces more build dependencies we can't get rid of (though users don't have to install those). 

On the other hand, I see your point about running tests with `ctest`.  Seems possible to still do that even if we use `pytest`, maybe along the lines of the suggestion in this SO [answer.](https://stackoverflow.com/a/52496856).  I'm not sure though, if that approach gives us the same flexibility to add/skip specific tests based on cmake configuration options.

Reply:
> On the other hand, I see your point about running tests with ctest. Seems possible to still do that even if we use pytest, maybe along the lines of the suggestion in this SO [answer.](https://stackoverflow.com/a/52496856). I'm not sure though, if that approach gives us the same flexibility to add/skip specific tests based on cmake configuration options

One problem with this is pytest scan the directories and execute every file matching test_*.py . This will be hard to translate to ctest where each cmake test call a single test file. Note that also many tests use mpirun which it will be difficult to translate to pytest.